### ISSUE TITLE
Go to definition goes to `lib.es6.d.ts` definition instead of local class

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#12866]
From https://github.com/Microsoft/vscode/issues/17083

**TypeScript Version:**  2.1.4

**Code**

```ts
class History {
    foo() {
        new History()
    }
}
```

Trigger go to definition on `History` in `new History()`

**Expected behavior:**
TypeScript server returns definition result for the `History` class defined in this file

**Actual behavior:**
Returns definition for `History` from `lib.es6.d.ts` instead.

**Comment by RyanCavanaugh** [comment]
This is a duplicate identifier error; there's no "local" class per se and it's just arbitrary which we go to. If this is in a module (add a top-level export / import statement), the go-to-def goes to the right place.

**Comment by mjbvz** [comment]
Yes, if this is in a TypeScript file then we see that error message. If the class is in a JavaScript file, I believe this is valid code and we do not generate any errors or warnings.

**Comment by mhegazy** [comment]
not sure there is much we can do here. we should consider showing this error in a .js (as part of https://github.com/Microsoft/TypeScript/issues/6802) file to make it clear that you are overriding a known global declaration.
